lgc/builder/DescBuilder.cpp

```diff
@@ -87,31 +87,51 @@ Value *DescBuilder::CreateLoadBufferDesc(unsigned descSet, unsigned binding, Val
       abstractType = ResourceNodeType::DescriptorBufferCompact;
 
     std::tie(topNode, node) = m_pipelineState->findResourceNode(abstractType, descSet, binding);
+    if (!node) {
+      // If we can't find the node, assume mutable descriptor and search for any node.
+      std::tie(topNode, node) = m_pipelineState->findResourceNode(ResourceNodeType::DescriptorMutable, descSet, binding);
+      if (!node) {
+        // We did not find the resource node. Return an undef value.
+        return UndefValue::get(getBufferDescTy(pointeeTy));
+      }
+    }
     assert(node && "missing resource node");
 
     if (node == topNode && isa<Constant>(descIndex) && node->concreteType != ResourceNodeType::InlineBuffer) {
-      // Handle a descriptor in the root table (a "dynamic descriptor") specially, as long as it is not variably
-      // indexed and is not an InlineBuffer. This lgc.root.descriptor call is by default lowered in
-      // PatchEntryPointMutate into a load from the spill table, but it might be able to "unspill" it to
-      // directly use shader entry SGPRs.
-      // TODO: Handle root InlineBuffer specially in a similar way to PushConst. The default handling is
-      // suboptimal as it always loads from the spill table.
-      Type *descTy = getDescTy(node->concreteType);
-      std::string callName = lgcName::RootDescriptor;
-      addTypeMangling(descTy, {}, callName);
-      unsigned dwordSize = descTy->getPrimitiveSizeInBits() / 32;
-      unsigned dwordOffset = cast<ConstantInt>(descIndex)->getZExtValue() * dwordSize;
-      if (dwordOffset + dwordSize > node->sizeInDwords) {
-        // Index out of range
-        desc = UndefValue::get(descTy);
-      } else {
+      // If the node is a mutable descriptor create call
+      if (node->concreteType == ResourceNodeType::DescriptorMutable) {
+        Type *descTy = getInt8Ty()->getPointerTo(ADDR_SPACE_BUFFER_FAT_POINTER);
+        std::string callName = lgcName::RootDescriptor;
+        addTypeMangling(descTy, {}, callName);
+        unsigned dwordSize = 2;
+        unsigned dwordOffset = cast<ConstantInt>(descIndex)->getZExtValue() * dwordSize;
         dwordOffset += node->offsetInDwords;
         dwordOffset += (binding - node->binding) * node->stride;
         desc = CreateNamedCall(callName, descTy, getInt32(dwordOffset), Attribute::ReadNone);
```

Comment:
The `node == topNode` in the condition, which means that we found the resource node at the top level: in userdata instead of in a descriptor set.

So... can this case really happen? I would hope not :)

Reply:
Yes this code isn't needed, removed